apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -11,21 +11,47 @@ import {
 } from '@cdo/apps/code-studio/progressRedux';
 import ProgressBubble from '@cdo/apps/templates/progress/ProgressBubble';
 import {levelType} from '@cdo/apps/templates/progress/progressTypes';
+import $ from 'jquery';
 
 const styles = {
-  headerContainer: {
-    // With our new bubble we don't want any padding above/below
-    display: 'flex',
-    flexDirection: 'row',
-    alignItems: 'center',
-    paddingLeft: 5,
-    paddingRight: 5,
+  container: {
     backgroundColor: color.lightest_gray,
     border: `1px solid ${color.lighter_gray}`,
     borderRadius: 5,
     height: 40,
-    marginLeft: 4,
-    marginRight: 4
+    position: 'relative',
+    overflow: 'hidden'
+  },
+  outer: {
+    position: 'absolute',
+    paddingLeft: 4,
+    paddingRight: 4,
+    height: '100%',
+    whiteSpace: 'nowrap'
+  },
+  inner: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: '100%'
+  },
+  headerVignette: {
+    width: '100%',
+    height: '100%',
+    position: 'absolute',
+    pointerEvents: 'none'
+  },
+  headerVignetteLeftRight: {
+    background:
+      'linear-gradient(to right, rgba(231, 232, 234, 1) 0%, rgba(231, 232, 234, 0) 20px, rgba(231, 232, 234, 0) calc(100% - 20px), rgba(231, 232, 234, 1) 100%)'
+  },
+  headerVignetteLeft: {
+    background:
+      'linear-gradient(to right, rgba(231, 232, 234, 1) 0%, rgba(231, 232, 234, 0) 20px'
+  },
+  headerVignetteRight: {
+    background:
+      'linear-gradient(to right, rgba(231, 232, 234, 0) calc(100% - 20px), rgba(231, 232, 234, 1) 100%)'
```

Comment:
Could we extract and reuse any of the vignette styles that get used across many components in this PR?

Reply:
This one is unique but I unified the others.